Client/Frontend/Browser/SearchLoader.swift

```diff
@@ -0,0 +1,81 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Shared
+import Storage
+
+private let URLBeforePathRegex = NSRegularExpression(pattern: "^https?://([^/]+/)", options: nil, error: nil)!
+
+// TODO: Swift current requires that classes extending generic classes must also be generic.
+// This is a workaround until that requirement is fixed.
+typealias SearchLoader = _SearchLoader<AnyObject, AnyObject>
+
+/**
+ * Shared data source for the SearchViewController and the URLBar domain completion.
+ * Since both of these use the same SQL query, we can perform the query once and dispatch the results.
+ */
+class _SearchLoader<UnusedA, UnusedB>: Loader<Cursor, SearchViewController> {
+    private let history: History
+    private let urlBar: URLBarView
```

Comment:
Yeah. The point of the Loader seemed to me to be to abstract away loading the data from the views that needed it. By tying the loader to a particular view, it feels like we've lost that to me (i.e. the loader knows more about who's listening than "that guy implements a LoaderListener protocol").